DIP 35 (#129)
efdevcon/DIPs: DIPs/DIP-35.md

This would make Devcon’ website decentralized, censorship resistant and privacy preserving, and would make Devcon’s digital efforts aligned with its values, while offering an web3 AI assistant to help attendees have an amazing experience__

## Abstract
Swarm is a decentralized storage and communication system that was initially incubated by the Ethereum Foundation.

Fair Data Society is an initiative for decentralized self-sovereign data commons.

BraianKnowsAI is a non-custodial AI assistant for performing transactions and search web3 info on a custom-built knowledge base, and a Swarm Foundation grantee.

This DIP follows the Devcon 2022 Swarm DIP and enhances it:

Hosting the Devcon website and agenda on Swarm/FairData Society and integrating it with BriankKnowsAI to have a web3 native AI assistant.

## Motivation & Rationale

Implementing this DIP would make Devcon’s digital efforts aligned with its principles and values:

The website hosting on the Swarm Network will be web3 native, censorship resistant and privacy preserving. The website will be in control of the Devcon team and not of a centralized provider. 

> Adding BrianKnowsAI will enhance the experience for attendees and virtual visitors aslike, as attendees will be able to ask questions on the agenda, i.e. “Brian, when is Vitalik speaking?” or “Brian, what’s account abstraction?”. It will also help attendees perform transactions helpful for purchses "Brian bridge 100DAI2 from Ethereum Mainnet to Polygon", "Brian, transfer 10DAI to vitalik.eth to pay them for dinner"

Reviewer: ```suggestion
Adding BrianKnowsAI will enhance the experience for attendees and virtual visitors aslike, as attendees will be able to ask questions on the agenda, i.e. “Brian, when is Vitalik speaking?” or “Brian, what’s account abstraction?”. It will also help attendees perform transactions helpful for purchases "Brian bridge 100DAI2 from Ethereum Mainnet to Polygon", "Brian, transfer 10DAI to vitalik.eth to pay them for dinner"
```

> Adding BrianKnowsAI will enhance the experience for attendees and virtual visitors aslike, as attendees will be able to ask questions on the agenda, i.e. “Brian, when is Vitalik speaking?” or “Brian, what’s account abstraction?”. It will also help attendees perform transactions helpful for purchases "Brian bridge 100DAI2 from Ethereum Mainnet to Polygon", "Brian, transfer 10DAI to vitalik.eth to pay them for dinner"

Reviewer: ```suggestion
Adding BrianKnowsAI will enhance the experience for attendees and virtual visitors alike, as attendees will be able to ask questions on the agenda, i.e. “Brian, when is Vitalik speaking?” or “Brian, what’s account abstraction?”. It will also help attendees perform transactions helpful for purchases "Brian bridge 100DAI from Ethereum Mainnet to Polygon", "Brian, transfer 10DAI to vitalik.eth to pay them for dinner"
```

---
DIP: 35
Title: Make Devcon censorship resistant, privacy preserving and smarter
Status: Draft
Themes: Virtual experience, Freeform, Purchases & ID
Tags: Software, Communications, Event Production, Event Operations
Instances: Devcon 7 
Authors: costanza@ethswarm.org
Resources Required: Operations support
Discussion: https://forum.devcon.org/t/devcon-2024-agenda-on-swarm-with-fair-data-society-and-brianknowsai/3357

> ---

Reviewer: ```suggestion
Created: 2024-02-21
---
```

## Summary of Proposal

> __Simple Summary__

Reviewer: ```suggestion
```

Hosting the Devcon website with the Agenda of DevCon on Swarm/FairData Society and integrating BrianKnowsAI. 
This would make Devcon’ website decentralized, censorship resistant and privacy preserving, and would make Devcon’s digital efforts aligned with its values, while offering an web3 AI assistant to help attendees have an amazing experience__

> ##Abstract

Reviewer: ```suggestion
## Abstract
```

Fair Data Society is an initiative for decentralized self-sovereign data commons.

BraianKnowsAI is a non-custodial AI assistant for performing transactions and search web3 info on a custom-built knowledge base, and a Swarm Foundation grantee.

This DIP follows the Devcon 2022 Swarm DIP and enhances it:

Hosting the Devcon website and agenda on Swarm/FairData Society and integrating it with BriankKnowsAI to have a web3 native AI assistant.

## Motivation & Rationale

Implementing this DIP would make Devcon’s digital efforts aligned with its principles and values:

The website hosting on the Swarm Network will be web3 native, censorship resistant and privacy preserving. The website will be in control of the Devcon team and not of a centralized provider. 

Adding BrianKnowsAI will enhance the experience for attendees and virtual visitors aslike, as attendees will be able to ask questions on the agenda, i.e. “Brian, when is Vitalik speaking?” or “Brian, what’s account abstraction?”. It will also help attendees perform transactions helpful for purchses "Brian bridge 100DAI2 from Ethereum Mainnet to Polygon", "Brian, transfer 10DAI to vitalik.eth to pay them for dinner"

> 
> 

Reviewer: ```suggestion
```

## Implementation

>  Yes, implementing the Agenda on Swarm has already been implemented at Devcon 6. The implementation went smoothly, however it has been done the day before the event, so this year we would like to have more time in order to deliver it as soon as possible. 

Reviewer: ```suggestion
Yes, implementing the Agenda on Swarm has already been implemented at Devcon 6. The implementation went smoothly, however it has been done the day before the event, so this year we would like to have more time in order to deliver it as soon as possible. 
```

The 2022 project had two components, an Exporter plugin and a Decentralized Frontend. More details about it can be found in the DIP 18.

In addition, this year we would like to host the main environment of the Agenda and Devcon website on the Swarm Network. The solution has been implemented on other situations like blogposts: https://fairdataprotocol.bzz.link/

BrianKnowsAI has been implemented at EthRome: all the relevant information have been uploaded to a specific knowledge base named "ETHRome" and the hackers were able to ask questions like:

"What are the Ethrome bounties?"
"How does the subgraphs in the graph work?"
"When is the Sismo workshop?"

Brian provides both an answer and the files it has used to generate that answer.

Eg.
Q:"What are the ethrome bounties?"
Brian: The ethrome bounties are prizes from sponsors like Sismo, Base, gnosis, Brian and so on.
Attached File: Hacker Manual - Bounties

> 
> 
> 
> 

Reviewer: ```suggestion
```

## Operational Requirements & Ownership

> __Please answer questions below:__

Reviewer: ```suggestion
```

1. What actions are required to implement the proposal at Devcon(nect)? 
If this DIP is approved, we need the design specifications to develop the Website and the Agenda. Ideally, we should have a couple of calls in order to prepare a plan for the implementation. 

Moreover, we will need the content, in order to publish the website and to feed Brian with the right knowledge base.

2. Who will be responsible for the proposal to be implemented effectively? The owner of the DIP, Costanza Gallo (costgallo / costanza@ethswarm.org)

3. What other projects could this proposal be integrated with? (Bonus points for collaboration across teams :))
This project is already an integration between Swarm Network, BrianKnowsAI.
In addition, it could be integrated with the ticketing platform

## Links & Additional Information

Swarm website https://www.ethswarm.org/

Swarm whitepaper https://www.ethswarm.org/swarm-whitepaper.pdf

Book of Swarm https://papers.ethswarm.org/p/book-of-swarm/

Bee Documentation https://docs.ethswarm.org/

> 
> 

Reviewer: ```suggestion
```
